openmc/tally_derivative.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import division
+
+import sys
+from numbers import Integral
+from xml.etree import ElementTree as ET
+
+from six import string_types
+
+import openmc.checkvalue as cv
+
+
+# "Static" variable for auto-generated TallyDerivative IDs
+AUTO_TALLY_DERIV_ID = 10000
+
+def reset_auto_tally_deriv_id():
+    global AUTO_TALLY_ID
+    AUTO_TALLY_DERIV_ID = 10000
+
+
+class TallyDerivative(object):
+    """A material perturbation derivative to apply to a tally.
+
+    Parameters
+    ----------
+    derivative_id : Integral, optional
+        Unique identifier for the tally derivative. If none is specified, an
+        identifier will automatically be assigned
+
+    Attributes
+    ----------
+    id : Integral
+        Unique identifier for the tally derivative
+    variable : str
+        Accepted values are 'density', 'nuclide_density', and 'temperature'
+    material : Integral
+        The perturubed material
+    nuclide : str
+        The perturbed nuclide. Only needed for 'nuclide_density' derivatives.
+        Ex: 'Xe-135'
+
+    """
+
+    def __init__(self, derivative_id=None):
```

Comment:
Would you be ok adding optional `variable`, `material`, and `nuclide` arguments to the constructor?